docs/public/_redirects

```diff
@@ -2,6 +2,10 @@
 # Avoid conflicts with the other Next.js apps hosted under https://mui.com/
 /base-ui/_next/* /_next/:splat 301
 
+# Fix links to other MUI products in deploy previews
+/base-ui/* /base-ui/:splat 200
+/* https://mui.com/:splat 301
+
```

Comment:
I wonder if we should add this:

1. It's not clear to me that it will work if we redirection /base-ui/ URLs
2. If it does work, then it sounds like something all repositories should have, e.g. MUI X
3. But then https://www.notion.so/mui-org/docs-infra-Change-docs-domain-0ce329e7ebc34aba8fbf4f3c5e13eb5d. it might quickly not have value with Base UI in a different URL location.

So maybe

```suggestion
```

to keep this PR a bit more focused

Reply:
I did this to avoid reverting https://github.com/mui/base-ui/pull/321. Since we now reference routes from the monorepo, we can't control them.
This will be less important when we publish the site under mui.com, but for now, as deployment previews are the only public docs we have, IMO, it's better to have the links working.
Besides, this way we can try out it if there are any issues with this approach.

Reply:
Interesting PR (#321), in MUI X/Toolapd the read on this problem was that it was ok. A 404 is clearer than a URL change to figure what MUI page is hosted where.

Now, having these links working helps ensure that once the preview is deployed to production, ahrefs weekly crawl won't find 404 links, so fair enough.

If https://github.com/mui/mui-x/blob/41fb8d93c52eb97557e41c7784a22ac901e06846/docs/public/_redirects#L40 like redirections work, 👍 to propagate this to all the docs-infra consumers.

Reply:
It seems to work. I added a test redirect in https://github.com/mui/base-ui/pull/326/commits/d6e1c2c81c407264a4b1b87938ce5eddf048cc77 and the site redirects me as expected: https://662678144d579d0008951e2b--base-ui.netlify.app/base-ui/test-redirect/

Reply:
Nice. We will see how this behaves once integrated in mui.com 👍.

I wonder how we will plan the mui.com/base-ui/ to base-ui.mui.com/ migration.